python/paddle/v2/framework/optimizer.py

```diff
@@ -63,7 +67,7 @@ def _finish_update(self, block):
         """
         pass
 
-    def _add_accumulator(self, block, name, param, dtype=None, fill_value=0.0):
```

Comment:
We should not remove dtype. It is not necessary that the type of the accumulator will be the same as that of param. dtype should be an argument to the function. There can be cases where we want integer accumulators for float parameters.

Reply:
OK.
```
if dtype is None:
   dtype = param.data_type
```

Reply:
Not really. If dtype is None, then let it be float, which is the default when we create variables.

Reply:
ok, I will change this

Reply:
done

Reply:
> Not really. If dtype is None, then let it be float, which is the default when we create variables. dtype should be independent of the param type.

We cannot use `float` for every accumulator. If the parameter is double, the accumulator should be double.

Reply:
Yes that sounds like a valid point. @jacquesqiao could you please change it what @reyoung has suggested. I apologize for the rework that you have to do.

Reply:
done